selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/AskForValueDialog.java

```diff
@@ -105,90 +108,88 @@ public AskForValueDialog(
     this.seleniumWebDriver = seleniumWebDriver;
     this.loader = loader;
     this.actionsFactory = actionsFactory;
+    this.loadPageWait = new WebDriverWait(seleniumWebDriver, LOAD_PAGE_TIMEOUT_SEC);
+    this.redrawElementWait = new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC);
     PageFactory.initElements(seleniumWebDriver, this);
   }
 
   public void waitFormToOpen() {
-    new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC)
-        .until(ExpectedConditions.visibilityOf(form));
+    redrawElementWait.until(visibilityOf(form));
   }
 
   public void waitFormToClose() {
-    new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC)
-        .until(ExpectedConditions.invisibilityOfElementLocated(By.id(ASK_FOR_VALUE_FORM)));
+    redrawElementWait.until(invisibilityOfElementLocated(By.id(ASK_FOR_VALUE_FORM)));
     loader.waitOnClosed();
   }
 
   /** click ok button */
   public void clickOkBtn() {
-    okBtn.click();
+    loadPageWait.until(visibilityOf(okBtn)).click();
     waitFormToClose();
   }
 
   /** click cancel button */
   public void clickCancelBtn() {
-    cancelBtn.click();
+    loadPageWait.until(visibilityOf(cancelBtn)).click();
     waitFormToClose();
   }
 
   public void clickCancelButtonJava() {
-    new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC)
-        .until(ExpectedConditions.visibilityOf(cancelButtonJava))
-        .click();
+    redrawElementWait.until(visibilityOf(cancelButtonJava)).click();
   }
 
   public void typeAndWaitText(String text) {
-    waitFormToOpen();
-    new WebDriverWait(seleniumWebDriver, TestTimeoutsConstants.LOAD_PAGE_TIMEOUT_SEC)
-        .until(ExpectedConditions.visibilityOf(input))
-        .sendKeys(text);
+    loadPageWait.until(visibilityOf(input)).sendKeys(text);
     waitInputNameContains(text);
   }
 
   public void waitNewJavaClassOpen() {
-    new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC)
-        .until(ExpectedConditions.visibilityOf(newJavaClass));
+    redrawElementWait.until(visibilityOf(newJavaClass));
   }
 
   public void waitNewJavaClassClose() {
-    new WebDriverWait(seleniumWebDriver, REDRAW_UI_ELEMENTS_TIMEOUT_SEC)
-        .until(ExpectedConditions.invisibilityOfElementLocated(By.id(NEW_JAVA_CLASS_FORM)));
+    redrawElementWait.until(invisibilityOfElementLocated(By.id(NEW_JAVA_CLASS_FORM)));
   }
 
   public void clickOkBtnNewJavaClass() {
-    okBtnnewJavaClass.click();
+    loadPageWait.until(visibilityOf(okBtnnewJavaClass)).click();
     loader.waitOnClosed();
   }
 
   public void typeTextInFieldName(String text) {
-    inputNameJavaClass.sendKeys(text);
+    loadPageWait.until(visibilityOf(inputNameJavaClass)).sendKeys(text);
+  }
+
+  public void waitTextInFieldName(String expectedText) {
```

Comment:
IMHO it's better to enclose call of this method into the **typeTextInFieldName()** in similar way, as it was done for **waitInputNameContains()** method.